### DISCUSSION TITLE
Custom Activities

### DISCUSSION THREAD

**Discussion by asaadhoussam** [elsa-workflows/elsa-guides#44] [Q&A]
I worked with Document Management Application(The sample project is called Document Management published by Sipke Schoorstra)
When I using the designer to create new workflow, I don't see the custom activities that were added in workflow/activities such as: Get Document, Archive Document, Zip File

help me please? thanks

**Comment by sfmskywalker** [comment]
Short answer: Studio will only show those custom activities if the **Elsa server** you are connected to registers the assembly that contains them. The designer does not discover activity classes from your project folder by itself.

What the current codebase shows:

- the usual registration pattern is `elsa.AddActivitiesFrom<Program>()` or `management.AddActivity<T>()`,
- `AddActivitiesFrom(...)` scans an assembly for concrete `IActivity` implementations and registers them,
- the current authoring docs also call out that new activities must be registered with management before they appear in the designer.

Concrete references:

- `elsa-samples/src/aspnet/Elsa.Samples.AspNet.WorkflowServer/Program.cs`
- `elsa-core/src/modules/Elsa.Workflows.Management/Extensions/ManagementServiceCollectionExtensions.cs`
- `elsa-core/doc/wiki/activities-and-authoring.md`

So if activities such as `Get Document`, `Archive Document`, and `Zip File` exist in your solution but do not show up in the designer, the first thing to check is whether the **server host** has registered that assembly.

In current code, that typically looks like:

```csharp
elsa.AddActivitiesFrom<Program>();
```

or registering the specific activity type explicitly.

One caveat: I could not find the old `Document Management` sample in the current maintained Elsa repos, so I can't point to that sample's exact startup file. If you are working from that older sample/article, look for the server project's Elsa registration and make sure the custom activity assembly is included there.

**Comment by sfmskywalker** [comment]
Closing this as answered. Custom activities only appear in Studio when the connected Elsa server registers the assembly or activity types, for example via `elsa.AddActivitiesFrom<...>()` or explicit management registration.
